### PR TITLE
mgmt: mcumgr: Consolidate Kconfig into a single file

### PR DESCRIPTION
**Pull request by carlescufi**
For some reason, MCUMgr used to consist of two separate Kconfig
files, likely due to the fact that the author's original intention was
to split the "management" and "mcumgr-specific" options in order to
integrate additional libraries. In practice this never worked out, so
coalesce them into a single file.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]
Nice improvement to kconfig menu.
Built for nr52840 and tested briefly, looks fine.

File: subsys/mgmt/mcumgr/Kconfig [hunk 1]
```diff
@@ -1,19 +1,250 @@
...
 config MCUMGR_SMP_BT
 	bool "Bluetooth mcumgr SMP transport"
-	select MCUMGR
 	select BT
 	select BT_PERIPHERAL
 	select BT_GATT_DYNAMIC_DB
 	help
 	  Enables handling of SMP commands received over Bluetooth.
 
+if MCUMGR_SMP_BT
```
Comment (mniestroj) [inline subsys/mgmt/mcumgr/Kconfig:247]: This is reduntant with `depends on MCUMGR_SMP_BT` in `MCUMGR_SMP_BT_AUTHEN`.

File: subsys/mgmt/mcumgr/Kconfig [hunk 2]
```diff
@@ -23,16 +254,18 @@ config MCUMGR_SMP_BT_AUTHEN
...
 	bool "Shell mcumgr SMP transport"
-	select MCUMGR
 	select SHELL
 	select SHELL_BACKEND_SERIAL
 	select BASE64
 	help
 	  Enables handling of SMP commands received over shell.  This allows
 	  the shell to be use for both mcumgr commands and shell commands.
 
+if MCUMGR_SMP_SHELL
```
Comment (mniestroj) [inline subsys/mgmt/mcumgr/Kconfig:268]: This is reduntant with `depends on MCUMGR_SMP_SHELL` in `MCUMGR_SMP_SHELL_MTU`.

File: subsys/mgmt/mcumgr/Kconfig [hunk 3]
```diff
@@ -53,6 +287,7 @@ config MCUMGR_SMP_UART
 	  commands to be received over UART without requiring an additional
 	  thread.
 
+if MCUMGR_SMP_UART
```
Comment (mniestroj) [inline subsys/mgmt/mcumgr/Kconfig:290]: This is redundant with `depends on MCUMGR_SMP_UART` in `MCUMGR_SMP_UART_MTU`.

File: subsys/mgmt/mcumgr/Kconfig [hunk 4]
```diff
@@ -75,6 +309,7 @@ config MCUMGR_SMP_UDP
 	  Enables handling of SMP commands received over UDP.
 	  Will start a thread for listening on the configured UDP port.
 
+if MCUMGR_SMP_UDP
```
Comment (mniestroj) [inline subsys/mgmt/mcumgr/Kconfig:312]: Please remove `depends on MCUMGR_SMP_UDP` in all options between `if MCUMGR_SMP_UDP` and `endif # MCUMGR_SMP_UDP`

File: subsys/mgmt/mcumgr/Kconfig [hunk 5]
```diff
@@ -155,6 +391,4 @@ config MCUMGR_BUF_USER_DATA_SIZE
 
 	  Note that CONFIG_NET_BUF_USER_DATA_SIZE must be at least as big as
 	  MCUMGR_BUF_USER_DATA_SIZE.
-
```
Comment (mniestroj) [inline subsys/mgmt/mcumgr/Kconfig:158]: I think it is more common in Kconfig files to always have one NL before and one NL after `if` and `endif` statements.

**Comment by de-nordic** [comment]
I agree with @mniestroj  that the redundant `if` and `depend` could be removed.

**Review by aunsbjerg — APPROVED** [review]

File: subsys/mgmt/mcumgr/Kconfig (hunk 1 above)
Comment (carlescufi) [reply subsys/mgmt/mcumgr/Kconfig]: Done, thanks!

File: subsys/mgmt/mcumgr/Kconfig (hunk 2 above)
Comment (carlescufi) [reply subsys/mgmt/mcumgr/Kconfig]: Done, thanks!

File: subsys/mgmt/mcumgr/Kconfig (hunk 3 above)
Comment (carlescufi) [reply subsys/mgmt/mcumgr/Kconfig]: Done, thanks!

File: subsys/mgmt/mcumgr/Kconfig (hunk 4 above)
Comment (carlescufi) [reply subsys/mgmt/mcumgr/Kconfig:306]: Done, thanks!

File: subsys/mgmt/mcumgr/Kconfig (hunk 5 above)
Comment (carlescufi) [reply subsys/mgmt/mcumgr/Kconfig:158]: I think this varies to be honest.

**Review by mniestroj — APPROVED** [review]

**Pull request merged**